lib/checkMergeConflicts.js

```diff
@@ -1,54 +1,85 @@
+  var hasMergeConflictLabel = false;
+  var pullRequestNumber = pullRequest.number;
+
+  // eslint-disable-next-line no-console
+  console.log(
+    'CHECKING PR NUMBER ' + pullRequestNumber + ' FOR MERGE CONFLICTS..');
+
+  // eslint-disable-next-line no-console
+  console.log(' IS PR MERGEABLE BEFORE POLLING? ' + pullRequest.mergeable);
+
+  // The value of the mergeable attribute can be true, false, or null.
+  // If the value is null, then GitHub has started a background job to compute the mergeability.
+  // After giving the job time to complete, resubmit the request.
+  // When the job finishes, a non-null value for the mergeable attribute will be observed in the response.
+  // Therefore, we poll the request until a non-null value is observed as suggested here:
+  // https://developer.github.com/v3/git/#checking-mergeability-of-pull-requests
+  while (pullRequest.mergeable === null) {
+    // Wait for 2 seconds before making another request.
+    await sleep(2000);
+    var pullRequestPromiseObj = await context.github.pullRequests.get(
       context.repo({number: pullRequestNumber}));
-    // eslint-disable-next-line no-console
-    console.log(
-      'CHECKING PR NUMBER ' + pullRequestNumber + ' FOR MERGE CONFLICTS..');
-    pullRequestDetails = pullRequestDetailsPromiseObj.data;
-    hasMergeConflictLabel = false;
-    labels = pullRequestDetails.labels;
-    for (var label in labels) {
-      if (labels[label].name === mergeConflictLabel[0]) {
-        hasMergeConflictLabel = true;
-        break;
-      }
-    }
+    pullRequest = pullRequestPromiseObj.data;
+  }
 
-    isMergeable = pullRequestDetails.mergeable;
-
-    if (hasMergeConflictLabel === false && isMergeable === false) {
-      userName = pullRequestDetails.user.login;
-      var linkText = 'link';
-      var linkResult = linkText.link(
-        'https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/');
-      var params = context.repo({
-        number: pullRequestNumber,
-        body: 'Hi @' + userName +
-            '. Due to recent changes in the "develop" branch, ' +
-            'this PR now has a merge conflict. ' +
-            'Please follow this ' + linkResult +
-            ' if you need help resolving the conflict, ' +
-            'so that the PR can be merged. Thanks!'});
-      labelPromiseObj = await context.github.issues.addLabels(context.repo({
-        number: pullRequestNumber,
-        labels: mergeConflictLabel}));
-      await context.github.issues.createComment(params);
-    }
+  // eslint-disable-next-line no-console
+  console.log(' IS PR MERGEABLE AFTER POLLING? ' + pullRequest.mergeable);
 
-    if (hasMergeConflictLabel === true && isMergeable === true) {
-      await context.github.issues.removeLabel(context.repo({
-        number: pullRequestNumber,
-        name: mergeConflictLabel[0]
-      }));
+  var labels = pullRequest.labels;
+  for (var label in labels) {
```

Comment:
`labels` here is a list, right? If so, you can do:
```
hasMergeConflictLabel = labels.some(function(label) {
  return label.name === mergeConflictLabel;
});
```

(assuming mergeConflictLabel is a string). 

As an aside, if labels is an array, it's misleading to call the dummy variable `label`. It should be `index`.

Reply:
Done, thanks!